src/app/ishop/product/product.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { ProductCardState, ProductInterface } from 'src/app/shared/interfaces';
+
+@Component({
+  selector: 'app-product',
+  templateUrl: './product.component.html',
+  styleUrls: ['./product.component.scss']
+})
+export class ProductComponent{
+  @Input('products-arr') productsArr: Array<ProductInterface> = [];
+  @Input('selected') selected!: number | null;
+  @Input() productCardState!: ProductCardState;
+
+  @Output('on-row-click') public onRowClick = new EventEmitter<number>();
+  @Output('on-delete') public onDelete = new EventEmitter<number>();
+  @Output('on-edit') public onEdit = new EventEmitter<number>();
+
+
+  // constructor() {
+  // }
+
+  handleRowClick(id: number | null) {
+    if (typeof id !== 'number') return;
```

Comment:
it's ok. also you can throw new Error here. for instance:
if (typeof id !== 'number') throw new Error('The argument ID must be a number');